### PR TITLE
Theme JSON schema: add heading/button key to color definition

### PR DESCRIPTION
**Pull request by t-hamano**
Follow-up on #49131
Similar to #55675

## What?

This PR adds heading/button key to color definition in theme.json schema.

## Why?
To allow developers to design theme.json with reference to the correct schema.

## Testing Instructions

Create the following JSON file locally, specifying the JSON schema changed by this PR.

```json
{
	"$schema": "https://raw.githubusercontent.com/WordPress/gutenberg/schema/theme-button-heading/schemas/json/theme.json",
	"version": 2
}
```

- You should see the `button` and `heading` keys listed in the `settings.color` property.
- You should see the `button` and `heading` keys listed in the `settings.block.{blockName}.color` property.
- The default value of `true` should be set when the `button` or `heading` key is selected.

capture

### CODE REVIEW

File: schemas/json/theme.json [hunk 1]
```diff
@@ -231,6 +231,16 @@
 							"description": "Allow users to set text colors.",
 							"type": "boolean",
 							"default": true
+						},
+						"heading": {
+							"description": "Allow users to set heading colors.",
```
Comment (carolinan) [inline schemas/json/theme.json]: I find this confusing because it can easily be mixed up with styles > elements > headings.

Comment (t-hamano) [reply schemas/json/theme.json]: Indeed. What about "Allow users to set heading colors **in a block**."?

**Review by ajlende — APPROVED** [review]
I'm fine with either phrasing for the description. If we do add "in a block" to the `heading` description, we should also update the description for `button`, `link`, and `text` too.

**Comment by t-hamano** [comment]
> If we do add "in a block" to the `heading` description, we should also update the description for `button`, `link`, and `text` too.

Makes sense. The theme.json schema's `settings.color.{text|link|heading|button}` all have the phrase "in a block". I would like to add similar phrases to these properties in block.json and then merge this PR.

block_json_description

**Pull request merged**